### PR TITLE
[url_launcher] Add support for setting show title on Chrome Custom Tabs

### PR DESCRIPTION
**Pull request by Alex-Usmanov**
- Adds showTitle option to platform interface & app-facing api
- Makes use of that parameter on android platform in order to set show title on Chrome Custom Tabs, when it is being used
- Makes android implementation use both legacy & new api (instead of passthrough from new to legacy), in order to allow use of the aforementioned parameter
- Adds a button to the example to test this parameter
- Adds tests

Screenshot_20231018-095928
(Notice title at the top of the page "CyLog Software - HTTP ...")

https://github.com/flutter/flutter/issues/136784

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [ ] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by Alex-Usmanov** [comment]
I made a terrible mistake with source control and #5166 was automatically closed. This basically #5166 in the latest condition it ever was.

I am so sorry for the inconvenience. This PR was reviewed and is in progress of landing. 

Platform interface portion: #5758
Platform implementation portion: #5759

**Comment by Alex-Usmanov** [comment]
This is blocked on #5759, which is currently in review.

**Comment by stuartmorgan** [comment]
@Alex-Usmanov Is this ready for re-review? The blocking PR has landed.

**Comment by Alex-Usmanov** [comment]
Yes, sir. It is ready for re-review

File: packages/url_launcher/url_launcher/lib/src/type_conversion.dart [hunk 1]
```diff
@@ -8,14 +8,24 @@ import 'types.dart';
...
+    WebViewConfiguration config) {
   return InAppWebViewConfiguration(
     enableJavaScript: config.enableJavaScript,
     enableDomStorage: config.enableDomStorage,
     headers: config.headers,
   );
 }
 
+/// Converts an (app-facing) [BrowserConfiguration] to a (platform interface)
+/// [InApp].
```
Comment (stuartmorgan) [inline packages/url_launcher/url_launcher/lib/src/type_conversion.dart:21]: `[InAppBrowserConfiguration]`

**Review by stuartmorgan — APPROVED** [review]
Sorry for the delay; LGTM with one comment nit.

@gmackall Could you do the secondary review here?

**Pull request merged**